rtengine/procparams.cc

```diff
@@ -4740,6 +4818,49 @@ int ProcParams::load(const Glib::ustring& fname, ParamsEdited* pedited)
                 colorToning.labgridBLow *= scale;
                 colorToning.labgridBHigh *= scale;
             }
+            std::vector<ColorToningParams::LabCorrectionRegion> lg;
+            bool found = false;
+            bool done = false;
+            for (int i = 1; !done; ++i) {
+                ColorToningParams::LabCorrectionRegion cur;
+                done = true;
+                std::string n = std::to_string(i);
+                if (assignFromKeyfile(keyFile, "ColorToning", Glib::ustring("LabRegionA_") + n, pedited, cur.a, pedited->colorToning.labregions)) {
+                    found = true;
+                    done = false;
+                }
+                if (assignFromKeyfile(keyFile, "ColorToning", Glib::ustring("LabRegionB_") + n, pedited, cur.b, pedited->colorToning.labregions)) {
+                    found = true;
+                    done = false;
+                }
+                if (assignFromKeyfile(keyFile, "ColorToning", Glib::ustring("LabRegionSaturation_") + n, pedited, cur.saturation, pedited->colorToning.labregions)) {
+                    found = true;
+                    done = false;
+                }
+                if (assignFromKeyfile(keyFile, "ColorToning", Glib::ustring("LabRegionLightness_") + n, pedited, cur.lightness, pedited->colorToning.labregions)) {
+                    found = true;
+                    done = false;
+                }
+                if (assignFromKeyfile(keyFile, "ColorToning", Glib::ustring("LabRegionHueMask_") + n, pedited, cur.hueMask, pedited->colorToning.labregions)) {
+                    found = true;
+                    done = false;
+                }
+                if (assignFromKeyfile(keyFile, "ColorToning", Glib::ustring("LabRegionChromaticityMask_") + n, pedited, cur.chromaticityMask, pedited->colorToning.labregions)) {
+                    found = true;
+                    done = false;
+                }
+                if (assignFromKeyfile(keyFile, "ColorToning", Glib::ustring("LabRegionLightnessMask_") + n, pedited, cur.lightnessMask, pedited->colorToning.labregions)) {
+                    found = true;
+                    done = false;
+                }
+                if (!done) {
+                    lg.emplace_back(cur);
```

Comment:
`emplace_back()` has no benefit here. Better use `push_back()` to make that clear.

Reply:
Maybe a bit odd: Couldn't those `if`s be compacted by

```c++
if (
    assignFromKeyfile(...)
    + assignFromKeyfile(...)
    + assignFromKeyfile(...)
    ...
) {
    found = true;
    done = false;
}
```

? That would at least make it a bit less verbose. Just an idea.